ocs_ci/utility/utils.py

```diff
@@ -882,33 +882,53 @@ class TimeoutSampler(object):
     Yielding the output allows you to handle every value as you wish.
 
     Feel free to set the instance variables.
+
+
+    Args:
+        timeout (int): Timeout in seconds
+        sleep (int): Sleep interval in seconds
+        func (function): The function to sample
+        func_args: Arguments for the function
+        func_kwargs: Keyword arguments for the function
     """
 
     def __init__(self, timeout, sleep, func, *func_args, **func_kwargs):
         self.timeout = timeout
-        """ Timeout in seconds. """
         self.sleep = sleep
-        """ Sleep interval seconds. """
         # check that given timeout and sleep values makes sense
         if self.timeout < self.sleep:
             raise ValueError("timeout should be larger than sleep time")
 
         self.func = func
-        """ A function to sample. """
         self.func_args = func_args
-        """ Args for func. """
         self.func_kwargs = func_kwargs
-        """ Kwargs for func. """
 
+        # Timestamps of the first and most recent samples
         self.start_time = None
-        """ Time of starting the sampling. """
         self.last_sample_time = None
-        """ Time of last sample. """
-
+        # The exception to raise
         self.timeout_exc_cls = TimeoutExpiredError
-        """ Class of exception to be raised.  """
-        self.timeout_exc_args = (self.timeout,)
-        """ An args for __init__ of the timeout exception. """
+        # Arguments that will be passed to the exception
+        self.timeout_exc_args = [self.timeout]
+        try:
+            self.timeout_exc_args.append(
+                f"Timed out after {timeout}s running {self._build_call_string()}"
+            )
+        except Exception:
+            log.exception(
+                "Failed to assemble call string. Not necessarily a test failure."
```

Comment:
what would cause this exception?

Reply:
Good question: hopefully nothing. But certain edge cases (like `func_kwargs` not being a `dict`) could happen. It would almost certainly be a regression introduced later on.

Often times in error reporting code, you want to be extra defensive against things like that. I wouldn't want a test run to die because someone slightly broke an error message. Make sense?

Reply:
:+1: